src/plugins/kibana/public/dashboard/index.js

```diff
@@ -115,9 +118,30 @@ define(function (require) {
             docTitle.change(dash.title);
           }
 
+          initPanelIndices();
           $scope.$emit('application.load');
         }
 
+        function initPanelIndices() {
+          // find the largest panelIndex in all the panels
+          var maxIndex = getPanelIndex();
+
+          // ensure that all panels have a panelIndex
+          $scope.state.panels.forEach(function (panel) {
+            if (!panel.panelIndex) {
+              panel.panelIndex = maxIndex++;
+            }
+          });
+        }
+
+        function getPanelIndex() {
```

Comment:
Renaming this function to `getMaxPanelIndex` or something similar may be helpful, as far as readability is concerned